lib/widgets/content.dart

```diff
@@ -255,18 +256,29 @@ class CodeBlock extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
-    final text = node.text;
-
     return Container(
-      padding: const EdgeInsets.fromLTRB(7, 5, 7, 3),
       decoration: BoxDecoration(
         color: Colors.white,
         border: Border.all(
           width: 1,
-          color: const HSLColor.fromAHSL(0.15, 0, 0, 0).toColor())),
+          color: const HSLColor.fromAHSL(0.15, 0, 0, 0).toColor()),
+        borderRadius: BorderRadius.circular(4)),
       child: SingleChildScrollViewWithScrollbar(
         scrollDirection: Axis.horizontal,
-        child: Text(text, style: _kCodeBlockStyle)));
+        child: Padding(
+          padding: const EdgeInsets.fromLTRB(7, 5, 7, 3),
+          child: Text.rich(_buildNodes(node.spans)),
+        )));
+  }
+
+  InlineSpan _buildNodes(List<CodeBlockSpanNode> nodes) {
+    return TextSpan(
+      style: _kCodeBlockStyle,
+      children: nodes.map(_buildNode).toList(growable: false));
+  }
+
+  InlineSpan _buildNode(CodeBlockSpanNode node) {
+    return TextSpan(text: node.text, style: codeBlockTextStyle(node.type));
```

Comment:
Let's also squash the commit that adds the styles:
fb44680c5 content: Add styles for each token type

into the commit that adds the parsing:
739faee0e content: Add CodeBlockSpanNode to parse span types

The parsing is only useful because of the styles, so I think it makes more sense to read that way.

Reply:
squashed